Local variable 'pex' will be copied despite being returned by name

### What is the issue?

This might be a bogus warning, but I'll still let you investigate.
It's a warning that happens with an old Xcode version (11.3.1) on an old macOS system. I see:
>Local variable 'pex' will be copied despite being returned by name  
>Call 'std::move' explicitly to avoid copying

On remove_bad_pex:
https://github.com/transmission/transmission/blob/4da965365c56c39d8fe19700f5f860ec496476dd/libtransmission/tr-dht.cc#L390

If the warning is bogus, you may close the issue: it's not visible in newer versions of Xcode (15.0.1).

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.x

On the other hand, latest Xcode is giving this warning on main branch:
```
./libtransmission/variant.h:146:57: warning: Forwarding reference passed to std::move(), which may unexpectedly cause lvalues to be moved; use std::forward() instead [bugprone-move-forwarding-reference]
            return { vec_.emplace_back(key, tr_variant{ std::move(val) }).second, true };
                                                        ^
```

--- a/libtransmission/tr-dht.cc
+++ b/libtransmission/tr-dht.cc
@@ -388,7 +388,7 @@ private:
         };
 
         pex.erase(std::remove_if(std::begin(pex), std::end(pex), IsBadPex), std::end(pex));
-        return pex;
+        return std::move(pex);
     }
 
     static void callback(void* vself, int event, unsigned char const* info_hash, void const* data, size_t data_len)

--- a/libtransmission/variant.h
+++ b/libtransmission/variant.h
@@ -143,7 +143,7 @@ public:
                 return { iter->second, false };
             }
 
-            return { vec_.emplace_back(key, tr_variant{ std::move(val) }).second, true };
+            return { vec_.emplace_back(key, tr_variant{ std::forward<Val>(val) }).second, true };
         }
 
         // --- custom functions

